lib/danger/ci_source/custom_ci_with_github.rb

```diff
@@ -0,0 +1,44 @@
+require "danger/request_sources/github/github"
+
+module Danger
+  # ### CI Setup
+  #
+  # You can use `danger/danger` Action in your `.github/workflows/xxx.yml`.
+  # And so, you can use GITHUB_TOKEN secret as `DANGER_GITHUB_API_TOKEN` environment variable.
+  #
+  #  ```yml
+  #  ...
+  #    steps:
+  #      - uses: actions/checkout@v3
+  #      - uses: danger/danger@master
+  #        env:
+  #          DANGER_GITHUB_API_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  #  ```
```

Comment:
These docs don't describe how to use this setup (via the env var: `CUSTOM_CI_WITH_GITHUB`) - it seems to be a copy of the docs for GitHub Actions support? https://github.com/danger/danger/blob/master/lib/danger/ci_source/github_actions.rb

Reply:
yes.  usage is identical.  only diff is the environment key.  there is opportunity for refactoring to a base class maybe.  is that preferred?

Reply:
I don't really care about making it a shared abstraction, they're shallow and have hard boundaries - I worry as someone reading the docs looking at this and thinking "which do I use for github actions?" because this to me looks like the GitHub Actions CI setup but wouldn't actually work

The docs need to explain what an "Custom CI" is, what env var needs to be set, and why you would use it instead of the github actions docs

Reply:
@orta apologies.  very bad copy paste previously.  inline docs updated.